File: Quivi.Backoffice.Api/Requests/ChannelProfiles/CreateChannelProfileRequest.cs
using Quivi.Domain.Entities.Pos;

namespace Quivi.Backoffice.Api.Requests.ChannelProfiles
{
    public class CreateChannelProfileRequest
    {
        public required string Name { get; init; } = string.Empty;
        public decimal MinimumPrePaidOrderAmount { get; init; }
        public ChannelFeature Features { get; init; }
        public TimeSpan? SendToPreparationTimer { get; init; }
        public required string PosIntegrationId { get; init; } = string.Empty;
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/channelProfiles/CreateChannelProfileRequest.ts
import { ChannelFeatures } from "./ChannelProfile";

export interface CreateChannelProfileRequest {
    readonly name: string;
    readonly minimumPrePaidOrderAmount: number;
    readonly features: ChannelFeatures;
    readonly sendToPreparationTimer?: string;
    readonly posIntegrationId: string;
}